test/models/compute_resources/vmware_test.rb

```diff
@@ -1072,4 +1064,47 @@ def mock_network(id, name, virtualswitch = nil)
       check_vm_attribute_names(cr)
     end
   end
+
+  describe '#generate_secure_boot_settings' do
+    before do
+      @cr = FactoryBot.build_stubbed(:vmware_cr)
+    end
+
+    test "returns secure boot settings when firmware is 'uefi_secure_boot'" do
+      assert_equal({ "secure_boot" => true }, @cr.send(:generate_secure_boot_settings, 'uefi_secure_boot'))
+    end
+
+    test "returns an empty hash for firmware types other than 'uefi_secure_boot'" do
+      assert_empty @cr.send(:generate_secure_boot_settings, 'uefi')
+      assert_empty @cr.send(:generate_secure_boot_settings, 'bios')
+      assert_empty @cr.send(:generate_secure_boot_settings, '')
+      assert_empty @cr.send(:generate_secure_boot_settings, nil)
+    end
+  end
+
+  describe '#validate_tpm_compatibility' do
+    before do
+      @cr = FactoryBot.build_stubbed(:vmware_cr)
+    end
+
+    test 'returns true and no errors when firmware is EFI and virtual_tpm is enabled' do
+      assert_equal true, @cr.send(:validate_tpm_compatibility, '1', 'efi')
+      assert_empty @cr.errors.full_messages
+    end
+
+    test 'returns false and no errors when firmware is EFI and virtual_tpm is disabled' do
+      assert_equal false, @cr.send(:validate_tpm_compatibility, '0', 'efi')
+      assert_empty @cr.errors.full_messages
```

Comment:
```suggestion
      assert_empty @cr.errors
```

Nit: Isn't this enough to check?